mcd/mcd-seth/mcd-seth.md

```diff
@@ -125,192 +168,274 @@ In order to better understand the MCD contracts, the following provides a brief
+export debtWadRound=$(bc<<<"($art*$rate*10^${WAD_DECIMALS})/1 + 1")
 ```
 
 - `art`: internal vault debt representation
 - `rate`: accumulated stability fee from the system
 - `debt`: vault debt in Dai
 - `debtWadRound`: vault debt in wad (i.e. multiplied by 10^18), added by 1 wad to avoid rounding issues.
 
-Then we need to approve the transfer of DAI tokens to the adapter. Call the approve function of the DAI ERC-20 token contract with the right parameters:
+Then we need to approve the transfer of DAI tokens to the adapter. Call the `approve` function of the DAI ERC-20 token contract with the right parameters:
 
-`seth send $DAI_TOKEN 'approve(address,uint256)' $MCD_JOIN_DAI $debtWadRound`
+```bash
+seth send $MCD_DAI 'approve(address, uint256)' $MCD_JOIN_DAI $debtWadRound
+```
 
-If we want to be sure that our approve transaction succeeded, we can check the results with this command:
+If we want to be sure that our approval transaction succeeded, we can check the results with this command:
 
-`seth --from-wei $(seth --to-dec $(seth call $DAI_TOKEN 'allowance(address, address)' $ETH_FROM $MCD_JOIN_DAI)) eth`
+```bash
+seth call $MCD_DAI 'allowance(address, address)(uint256)' $ETH_FROM $MCD_JOIN_DAI | seth --from-wei
+```
 
 Output:
 
-`20.000000000000000001`
+`15000.041850037339693452`
 
 Now to actually join the Dai to the adapter:
 
-`seth send $MCD_JOIN_DAI "join(address,uint)" $urn $debtWadRound`
+```bash
+seth send $MCD_JOIN_DAI "join(address, uint256)" $urn $debtWadRound
+```
 
 To make sure it all worked:
 
-`seth --to-fix 45 $(seth --to-dec $(seth call $MCD_VAT 'dai(address)(uint256)' $urn))`
+```bash
+seth call $MCD_VAT 'dai(address)(uint256)' $urn | seth --to-fix $RAD_DECIMALS
+```
 
 Output:
 
-`20.000000000000000001000000000000000000000000000`
+`15000.041850037339693452000000000000000000000000000`
 
-Now, onto actually getting our collateral back. `dart` and `dink`, as the d in their abbreviation stands for delta, are inputs for changing a value, and thus they can be negative. When we want to lower the amount of DAI drawn from the `urn`, we lower the art parameter of the `urn`. Again, we need to use the `frob` operation to change these parameters: `frob(uint cdpId, address ETH_FROM, int dink, int dart)`
+Now, onto actually getting our collateral back. `dart` and `dink`, as the `d` in their abbreviation stands for delta, are inputs for changing a value, and thus they can be negative. When we want to lower the amount of DAI drawn from the `urn`, we lower the art parameter of the `urn`.
 
 We only need to set up the `dink` and `dart` variables.
-
-Using [mcd-cli](https://github.com/makerdao/mcd-cli#installation), we create our two negative `$dink` and `$dart`.
+Using [mcd-cli](https://github.com/makerdao/mcd-cli#installation), we create our two negative `$dink` and `$dart`:
 
 ```bash
-dink=$(seth --to-uint256 $(mcd --to-hex $(seth --to-wei -150 eth)))
+dink=$(seth --to-uint256 $(mcd --to-hex $(seth --to-wei -5 eth)))
```

Comment:
I'd suggest to consider the following change as It seems that at the time of writing of this tutorial `seth` couldn't manage negative integers, that now seem to be supported.
The following should return the same output:
```suggestion
dink=$(seth --to-int256 $(seth --to-wei -5 eth))
```